doc/blog/201708-Importing-SL-Toolkit-part-1.md

```diff
@@ -0,0 +1,122 @@
+
+This project will contain all the XAML files and C# source files required for the `TreeView` to function properly. It uses the excellent [MSBuild.Sdk.Extras](https://github.com/onovotny/MSBuildSdkExtras) msbuild extensions to cross targeted library with minimal efforts, using the new and improved _sdk-style_ project format, and simplifies the creation of NuGet packages.
+
+It will then possible to create an installable NuGet package using the context menu [Pack option](https://docs.microsoft.com/en-us/nuget/quickstart/create-and-publish-a-package-using-visual-studio#run-the-pack-command) on the project.
+
+### Importing the source
+
+The process of importing the source is somewhat straightforward. Microsoft, in all its XAML variants, kept many of the APIs signature-compatible. This means that in a large majority of cases, [simply changing the namespaces](https://github.com/nventive/Uno.UI.Toolkit.SL/commit/d4da7a8ff33da6c9d45bebafa8c8ca65f6182612#diff-b35234eeeb3bdb81d82b850985bf37b3L9) from `System.Windows` to `Windows.UI` is making the code compatible with UWP.
+
+Here are some examples:
+- `System.Windows.Controls` -> `Windows.UI.Xaml.Controls`
+- `System.Windows.Input` -> `Windows.Devices.Input`
+- `System.Windows.Media` -> `Windows.UI.Xaml.Media`
+
+One tip here to simplify the migration is to temporarily remove all non-windows targets in the cross-targeted projects to keep only `uap10.0`. This helps in keeping the compilation errors limited to the UWP apis, and avoid some of the API differences that may happen including iOS/Android/Wasm targets. Once the Windows target builds, adding back the other targets will allow for special adjustments, if any.
+
+Most of the exercise of the code import is about making an heavy use of the Intellisense by importing files one by one, starting by `TreeView.cs` and removing all the red squiggles. The `TreeView` controls uses `TreeViewItem`, which in turn uses `HeaderedItemsControl`, etc...
+
+After the first pass of changing namespaces, and importing dependent files, we end up with a self-contained set of C# source files, but not yet compiling.
+
+### Adjusting for slightly changed APIs
+
+When trying to resolve the API differences when moving from UWP, after adjusting the namespaces, the first low hangings are the ones that are slightly different. For instance, it can be member visibility differences, or just name updates.
+
+Some examples:
+- `FrameworkElement.OnApplyTemplate` has been moved from `public` to `protected`
+- The `PropertyMetadata` does not have constructor that only contains a `PropertyChangedCallback` parameter
+- `GeneralTransform.Transform` is called `GeneralTransform.TransformPoint`
+- `Binding` does not contain a constructor taking a string path as a parameter
+- `Control.Focus` now requires a `FocusState` parameter
+
+Those a pretty easy to adjust, and the UWP runtime behavior has a great change of being identical to the one Silverlight had.
```

Comment:
"the UWP runtime behavior has a great change of being identical to the one Silverlight had."
How so? Maybe add a link here to further information.